### ISSUE TITLE
[V1] EdgeJS: Getting "Module did not self-register" error when runtime triggers a restart when using the edge node module.

### ISSUE THREAD

**Issue by ahmelsayed** [Azure/azure-functions-host#3074]
_From @eHealthManny on June 28, 2018 21:49_

I have a Javascript timer trigger running in azure. Everything works fine untill there is a "soft" restart of the function (ex: the function changes in the portal and is saved OR the function changes in my IDE locally). After this happens, every time my function triggers it crashes on the line:

`var edge = require('edge');`

"hard" restarting the function (clicking the restart button in the portal or closing my cmd window and restarting locally) fixes the issue until another soft restart happens. Here is the stack trace I get:

Locally: 
```
mscorlib: Exception while executing function: Functions.SomeFunc. mscorlib: One or more errors occurred. Error: Module did not self-register.
   at Error (native)
    at Object.Module._extensions..node (module.js:597:18)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (C:\MYAPP\node_modules\edge\lib\edge.js:54:8)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (C:\MYAPP\SomeFunc\index.js:22:12)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at eval (eval at compileFunc (C:\Users\my.user\AppData\Roaming\nvm\v6.12.2\node_modules\azure-functions-core-tools\bin\edge\double_edge.js:34:28), <anonymous>:1:80)
    at compileFunc (C:\Users\my.user\AppData\Roaming\nvm\v6.12.2\node_modules\azure-functions-core-tools\bin\edge\double_edge.js:35:16).
```

Azure
```
2018-06-28T21:40:30.009 [Error] Exception while executing function: Functions.SomeFunc. mscorlib: One or more errors occurred. Error: Module did not self-register.
    at Error (native)
    at Object.Module._extensions..node (module.js:597:18)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (D:\home\site\wwwroot\node_modules\edge\lib\edge.js:54:8)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (D:\home\site\wwwroot\SomeFunc\index.js:22:12)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at eval (eval at compileFunc (D:\Program Files (x86)\SiteExtensions\Functions\1.0.11820\bin\edge\double_edge.js:34:28), <anonymous>:1:80)
    at compileFunc (D:\Program Files (x86)\SiteExtensions\Functions\1.0.11820\bin\edge\double_edge.js:35:16).
```
It seems like it may have something to do with azure-functions-core-tools also using edge. Any help is much appreciated.

_Copied from original issue: Azure/azure-functions-core-tools#560_

**Comment by eHealthManny** [comment]
Some more information:
Node version Local : 6.12.2  
Node version Azure : 6.12.3
Edge version's tested 6.5.1 AND 7.10.1

My azure function looks like this:
```
var edge = require('edge');
module.exports = function (context) {
	context.log('Logging');
	context.done();
};
```

**Comment by mathewc** [comment]
Likely this is related to https://github.com/Azure/azure-functions-host/issues/2367. While that issue is about Edge and how we load it in our AppDomain, it points out how Edge performs AppDomain level registrations that get messed up in some restart scenarios.

**Comment by mhoeger** [comment]
Closing as won't fix, see #2467
